PROTOCOL.md

```diff
@@ -342,17 +346,19 @@ dataChange | Boolean | When `false` the logical file must already be present in
 stats | [Statistics Struct](#Per-file-Statistics) | Contains statistics (e.g., count, min/max values for columns) about the data in this logical file | optional
 tags | Map[String, String] | Map containing metadata about this logical file | optional
 deletionVector | [DeletionVectorDescriptor Struct](#Deletion-Vectors) | Either null (or absent in JSON) when no DV is associated with this data file, or a struct (described below) that contains necessary information about the DV that is part of this logical file. | optional
+baseRowId | Long  | Default generated Row ID of the first row in the file. The default generated Row IDs of the other rows in the file can be reconstructed by adding the physical index of the row within the file to the base Row ID. See also [Row IDs](#row-ids) | required
```

Comment:
required only when enableRowIds is enabled?

Reply:
Right. I'll change it to optional here since this is describing the schema of the `add` action (same for `remove`). The writers requirements for row ids are explicit about when the value must be set.